packages/platform-ios/src/config/findPodfilePath.ts

```diff
@@ -9,13 +9,43 @@
 import glob from 'glob';
 import path from 'path';
 
-export default function findPodfilePath(folder: string) {
-  const podfiles = glob.sync('**/Podfile', {
-    cwd: folder,
-    ignore: 'node_modules/**',
-  });
+// Regexp matching all test projects
+const TEST_PROJECTS = /test|example|sample/i;
+
+// Base iOS folder
+const IOS_BASE = 'ios';
+
+// These folders will be excluded from search to speed it up
+const GLOB_EXCLUDE_PATTERN = ['**/@(Pods|node_modules|Carthage)/**'];
+
+export default function findPodfilePath(cwd: string) {
+  /**
+   * First, we're going to look for all Podfiles within the `cwd`
+   */
+  const podfiles = glob
+    .sync('**/Podfile', {
+      cwd,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    /**
+     * Then, we will run a simple test to rule out most example projects,
+     * unless they are located in a `ios` folder
+     */
+    .filter(
+      (project) =>
+        path.dirname(project) === IOS_BASE || !TEST_PROJECTS.test(project),
+    )
+    /**
+     * Podfile from `ios` folder will be picked up as a first one.
+     */
+    .sort((project) => (path.dirname(project) === IOS_BASE ? -1 : 1));
+
   if (podfiles.length > 0) {
-    return path.join(folder, podfiles[0]);
+    /**
+     * @todo
+     * Shall we print a warning when multiple `Podfile` were found?
```

Comment:
I ran this locally and got:

```
> require("glob").sync('**/Podfile', { ignore: ['**/@(Pods|node_modules|Carthage)/**'] })
[ 'ios/Podfile', 'macos/Podfile' ]
```

I think printing a warning can be very helpful, but there can be false positives if you also have a macOS project. Maybe could print the list if `--verbose` is specified?

Reply:
I like this idea.

Reply:
We will favour `iOS` anyway since this is `iOSConfig` used for `run-ios`. But I guess we can adapt it in a later PR to work for `Mac` too?

Can you point me to a configuration/helpers that you use for `macos`?

Reply:
`findXcodeProject` is here: https://github.com/microsoft/react-native-macos/blob/main/local-cli/runMacOS/findXcodeProject.js

It only gets called from `run-macos` ([code](https://github.com/microsoft/react-native-macos/blob/main/local-cli/runMacOS/runMacOS.js)), so it can assume that `pod install` has happened. You wouldn't be able to use it for `config`.

Reply:
BTW, do you want me to export you this helper so you can reduce some code overlap? Or prefer to keep it internal?

Reply:
@HeyImChris, @Saadnajmi: What do you think?

Reply:
What's the exact question? What should we do if the CLI finds two podfiles (One for iOS and one for macOS)? I think that's a valid case (though the macOS podfile will have both an iOS and a macOS target in it I believe).

Reply:
I think @grabbou was asking if it makes sense for react-native-macos to reuse this function instead of the existing `findXcodeProject`. If so, he will make it public.

Reply:
What makes searching for podfiles better than searching for Xcode projects? They'll likely be paired 1:1 in most cases

Reply:
Question was whether you would like us to export `findXcodeProject` function from the `platform-ios` package, so you don't have to re-implement it yourself?

Reply:
Ah. If we do that, then `react-native-macos` takes a dependency on `@react-native-community/cli`? My initial thought is we wouldn't want that... but code sharing sounds good. 
Dumb question, does `react-native` currently take any dependency on `@react-native-community/cli`?

Reply:
@HeyImChris I believe we search for podfiles rather than Xcode projects to support scenarios like react-native-test-app, which generates an xcode project given a podfile. In effect, the source of truth becomes the podfile and not the xcode project. I remember when @tido64 pushed for this change in the CLI which in turn helped us use react-native-test-app for FluentUI React Native.

Reply:
@Saadnajmi there's a sub-package `@react-native-community/platform-ios`. React Native depends on it. CLI doesn't. CLI detects runtime packages with platforms (such as `platform-ios`, or `react-native-windows`) and adds commands / configurations based on installed packages.

So by depending on `platform-ios`, you wouldn't pull the entire CLI.

Reply:
I think this change makes sense given that we wouldn't be taking on a bunch of new dependencies with the obvious value of reducing our diffs. @Saadnajmi  sound good to you?

Reply:
As it turns out, we already have a [dependency on the CLI](https://github.com/microsoft/react-native-macos/blob/26a7c3e9d430ebb58b9c9e2aa220fe9019ae1e1f/package.json#L97) (thanks @HeyImChris for pointing it out) so yeah this would be great! Yay code sharing!

Reply:
Right -> https://github.com/microsoft/react-native-macos/blob/26a7c3e9d430ebb58b9c9e2aa220fe9019ae1e1f/package.json#L99.


I will open a follow-up issue to explore various code-sharing opportunities.